### DISCUSSION TITLE
Use alpha in the `theme` function

### DISCUSSION THREAD

**Discussion by u3u** [ben-rogerson/twin.macro#712] [Ideas]
Now there seems to be no way to specify alpha when getting the color through the `theme` function

```ts
import { theme } from 'twin.macro'

theme`colors.red.500` // #ef4444
```

Can you consider the following usage?

```ts
theme`colors.red.500/50` // rgba(239, 68, 68, 0.5)
```

**Comment by ben-rogerson** [comment]
that looks like a good improvement to me.

Some notes/questions:

- The theme value is due for a 'resync' with tailwind - investigation required
- I'm guessing the alpha value looks at the opacities from your theme - like the slash opacity in `tw` classes 
- Does `theme` have an arbitrary opacity value? eg: ```theme`colors.red.500/[.045]` ```
- Value should support spaces around the slash opacity, eg: ```theme`colors.red.500 / 50` ```
- [Here's where](https://github.com/ben-rogerson/twin.macro/blob/f8baa5e8e646f425d44a138def72c01f1fe68167/src/utils/misc.js#L67) these feature(s) will be patched in

**Reply by u3u** [reply]
> Does `theme` have an arbitrary opacity value? eg: `` theme`colors.red.500/[.045]`  ``

It would be nice if it could support arbitrary values!

> Value should support spaces around the slash opacity, eg: `` theme`colors.red.500 / 50`  ``

Yes, the only usage found in the [official documentation](https://tailwindcss.com/docs/functions-and-directives#theme) is something like `theme(colors.blue.500 / 75%)`

The usage that comes to mind is this:

```ts
theme`colors.red.500/50` // rgba(239, 68, 68, 0.5)
theme`colors.red.500/[.045]` // rgba(239, 68, 68, 0.045)

// Or add spaces, I don't know which is better, or support both?
theme`colors.red.500 / 50` // rgba(239, 68, 68, 0.5)
theme`colors.red.500 / [.045]` // rgba(239, 68, 68, 0.045)

// Official Usage
theme`colors.red.500 / 50%` // rgb(239 68 68 / 50%)
```

**Reply by ben-rogerson** [reply]
Spec looks good to me 👍 
Not sure when I'll be able to get around to it - perhaps after the v3 release.

**Reply by lightyen** [reply]
I think percent value or floating value is good enough, the key `50` is ambiguous.

```js
// tailwind.config.js
module.exports = {
  theme: {
    colors: {
      primary: 'rgb(var(--color-primary) / <alpha-value>)',
      red: 'hsl(0 100% 50% / <alpha-value>)',
    },
  },
}
```
```js
theme`colors.primary/50%` // rgb(var(--color-primary) / 50%)
theme`colors.primary/[.05]` // rgb(var(--color-primary) / 0.05)
theme`colors.red / 50%` // hsl(0 100% 50% / 50%)
theme`colors.red / [ .05 ]` // hsl(0 100% 50% / 0.05)
theme`colors.red` //  hsl(0 100% 50% / 1)

// But I prefer the form without brackets:
theme`colors.primary/50.7%` // rgb(var(--color-primary) / 50.7%), why not?
```

**Reply by ben-rogerson** [reply]
Great stuff here and I agree the slash opacity number is ambiguous 👍 

In the case of colour classes, when an incorrect slash opacity is added, twin displays an error message like this:

```bash
The slash opacity “25s” isn’t a valid opacity for “bg-black/253”
Fix: Try one of these opacity values:

- 0 > 0
- 5 > 0.05
- 10 > 0.1
// ...
``` 

I'd really like to keep the slash opacity syntax as synced as possible for the theme value.
So we could align the features by matching opacity values from their tailwind config.

Custom values (including percentages) could be added within an arbitrary opacity, eg:

```js
// tailwind.config.js
module.exports = {
  theme: {
    colors: {
      red: 'hsl(0 100% 50%)',
    },
  },
}

// ---

theme`colors.red/[50%]`
// ↓ ↓ ↓ ↓ ↓ ↓
"hsl(0 100% 50% / 50%)"
```

**Reply by lightyen** [reply]
AFAIK, tailwind doesn't support `colors.red.500/[0.82]` in theme function. https://play.tailwindcss.com/M832DP91GW

**Comment by dlopezjoffre-thesuite** [comment]
Hey, with tailwind v3 the `theme` function is completely broken! Tailwind will append the "/ <alpha-value>" part to all rgb color string on the theme (the ones defined with a var) and using the `theme` function will not replace that portion of the value with any number (not even 1 by default) so the color won't work anymore given it cannot be interpreted.

For example calling background: theme`colors.black`, black is returned raw from the config as `rgb(var(--black) / <alpha-value>)` and since the theme function does nothing with the alpha value, the css ends up as `backgroun: rgb(var(--black) / <alpha-value>)` which won't do anything given it's invalid.

@ben-rogerson Is there a plan to implement this idea? Or at least the possibility of a fix so colors (without an alpha) can work as they did with tailwind v2? Thanks!
